src/main/scala/beam/utils/plan/sampling/AvailableModeUtils.scala

```diff
@@ -50,81 +51,89 @@ object AvailableModeUtils extends LazyLogging {
 
   /**
     * Gets the excluded modes set for the given person
+    *
     * @param person the respective person
     * @return
     */
-  def availableModesForPerson(person: Person): Seq[BeamMode] = {
-    person.getCustomAttributes
-      .get(PopulationAdjustment.BEAM_ATTRIBUTES)
-      .asInstanceOf[AttributesOfIndividual]
-      .availableModes
-  }
+  def availableModesForPerson(person: Person): Seq[BeamMode] = gerPersonCustomAttributes(person).availableModes
 
   /**
     * Sets the available modes for the given person in the population
+    *
     * @param population population from the scenario
     * @param person the respective person
     * @param permissibleModes List of permissible modes for the person
     */
   def setAvailableModesForPerson(person: Person, population: Population, permissibleModes: Seq[String]): Unit = {
-    val attributesOfIndividual = person.getCustomAttributes
-      .get(PopulationAdjustment.BEAM_ATTRIBUTES)
-      .asInstanceOf[AttributesOfIndividual]
+    val attributesOfIndividual = gerPersonCustomAttributes(person)
     setModesForPerson(person, population, permissibleModes, attributesOfIndividual)
   }
 
-  def setModesForPerson(
+  private def setModesForPerson(
     person: Person,
     population: Population,
     permissibleModes: Seq[String],
     attributesOfIndividual: AttributesOfIndividual
   ): Unit = {
-    val excludedModes = getExcludedModesForPerson(population, person.getId.toString)
-    val availableModes = if (excludedModes.nonEmpty) {
-      permissibleModes.filterNot(am => excludedModes.exists(em => em.equalsIgnoreCase(am)))
-    } else {
-      permissibleModes
-    }
+    val availableModes = getAvailableModesOfPerson(person, population, permissibleModes)
     try {
-      person.getCustomAttributes
-        .put(
-          PopulationAdjustment.BEAM_ATTRIBUTES,
-          attributesOfIndividual.copy(availableModes = availableModes.map(f => BeamMode.withValue(f.toLowerCase)))
-        )
+      val attributesUpdated =
+        attributesOfIndividual.copy(availableModes = availableModes.map(f => BeamMode.withValue(f.toLowerCase)))
+      person.getCustomAttributes.put(PopulationAdjustment.BEAM_ATTRIBUTES, attributesUpdated)
     } catch {
       case e: Exception =>
         logger.error("Error while converting available mode string to respective Beam Mode Enums : " + e.getMessage, e)
     }
   }
 
+  private def getAvailableModesOfPerson(
+    person: Person,
+    population: Population,
+    permissibleModes: Seq[String]
+  ): Seq[String] = {
+    val excludedModes = getExcludedModesForPerson(population, person.getId.toString)
+    permissibleModes.filterNot(am => excludedModes.exists(em => em.equalsIgnoreCase(am)))
```

Comment:
You lost the existence check on `excludedModes`. I assume this was a micro-optimization, but it is more optimized. It should probably stay

Reply:
In this case outcome is exactly the same. Benefits: smaller code with at same level of readability(IMO is better readability) and micro optimization.